src/sage/schemes/curves/affine_curve.py

```diff
@@ -1800,22 +1839,40 @@ def fundamental_group(self, simplified=True, puiseux=False):
         In the case of number fields, they need to have an embedding
         to the algebraic field::
 
-            sage: # needs sage.rings.number_field
-            sage: a = QQ[x](x^2 + 5).roots(QQbar)[0][0]
+            sage: # needs sage.rings.number_field sirocco
+            sage: T.<t> = QQ[]
+            sage: a = (t^2 + 5).roots(QQbar)[0][0]
             sage: F = NumberField(a.minpoly(), 'a', embedding=a)
             sage: F.inject_variables()
             Defining a
             sage: A.<x,y> = AffineSpace(F, 2)
             sage: C = A.curve(y^2 - a*x^3 - x^2)
-            sage: C.fundamental_group()         # optional - sirocco
+            sage: C.fundamental_group()
             Finitely presented group < x0 |  >
+            sage: C = A.curve(x * (x - 1))
+            sage: C.fundamental_group()
+            Finitely presented group < x0, x1 |  >
 
         .. WARNING::
 
             This functionality requires the sirocco package to be installed.
         """
         from sage.schemes.curves.zariski_vankampen import fundamental_group_from_braid_mon
-        return fundamental_group_from_braid_mon(self.braid_monodromy(), simplified=simplified, puiseux=puiseux)
+        bm = self.braid_monodromy()
+        if bm == []:
```

Comment:
```suggestion
        if not bm:
```
A micro-optimization.

Reply:
Sometimes braid monodromy can be an empty list, say some non vertical parallel lines, and it would be different to `None`, let me think about it.

Reply:
The attributes are initialized to `None` and an empty list is a possible braid monodromy.

Reply:
But the output of `braid_monodromy()` could be `None`? That would be strange to me.

Reply:
You are right, changed